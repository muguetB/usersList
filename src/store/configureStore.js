// @flow

import createSagaMiddleware from 'redux-saga'
import { applyMiddleware, combineReducers, createStore, compose } from 'redux'
import createReducers from '../redux'
import sagas from '../sagas'

export default function configureStore() {
  const sagaMiddleware = createSagaMiddleware()

  const initialState =
    window &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()

  const middleware = applyMiddleware(sagaMiddleware)

  const appReducer = combineReducers(createReducers())

  const rootReducer = (state, action) => appReducer(state, action)

  const store = createStore(rootReducer, initialState, compose(middleware))

  sagas.forEach(saga => saga.map(sagaMiddleware.run))

  return store
}
