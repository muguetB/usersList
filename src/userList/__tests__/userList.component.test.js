// @flow
import 'react-native'
import React from 'react'
import UserList from '../userList.componet'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'

describe('UserList component', () => {
  it('should render correctly', () => {
    // given
    const props = {
      onPageChange: jest.fn(),
      itemsPerPage: 5,
      users: [],
      loading: false,
    }

    const mockStore = configureStore()({
      usersList: { users: [], loading: false, error: false, emptyUsersList: false },
    })

    // when
    const wrapper = renderer.create(
      <Provider store={mockStore}>
        <UserList {...props} />
      </Provider>,
    )

    // then
    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
