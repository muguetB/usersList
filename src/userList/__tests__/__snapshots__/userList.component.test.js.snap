// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList component should render correctly 1`] = `
<View>
  <Text>
    Waiting for data
  </Text>
</View>
`;
